conans/test/unittests/tools/microsoft/test_msvs_toolset.py

```diff
@@ -0,0 +1,68 @@
+import pytest
+from conan.tools.microsoft import msvs_toolset
+from conans.test.utils.mocks import ConanFileMock, MockSettings
+
+
+@pytest.mark.parametrize("compiler_version,expected_toolset", [
+    ("17", "v143"),
+    ("16", "v142"),
+    ("15", "v141"),
+    ("14", "v140"),
+    ("12", "v120"),
+    ("11", "v110"),
+    ("10", "v100"),
+    ("9", "v90"),
+    ("8", "v80")])
+def test_visual_studio_default(compiler_version, expected_toolset):
+    conanfile = ConanFileMock()
+    conanfile.settings = MockSettings({"compiler": "Visual Studio", "compiler.version": compiler_version})
+    assert expected_toolset == msvs_toolset(conanfile)
+
+
+@pytest.mark.parametrize("compiler_version,expected_toolset", [
```

Comment:
Better if 3 arguments, ``compiler_version, toolset, expected_toolset``, to understand the inputs and outputs and visually see the expected equals the input toolset (even a 1 line comment saying it would be nice)

Reply:
Done.